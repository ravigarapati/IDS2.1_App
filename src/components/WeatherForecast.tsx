import React from 'react';
import {View, Image, StyleSheet} from 'react-native';
import {CustomText} from './../components';

type WeatherForecastProps = {
  day: string;
  temperature: string;
  icon: string;
  accessibilityLabelDay?: string;
  accessibilityLabelTemperature?: string;
  accessibilityLabelImage?: string;
};

export default function WeatherForecast({
  day,
  temperature,
  icon,
  accessibilityLabelDay = '',
  accessibilityLabelTemperature = '',
  accessibilityLabelImage = '',
}: WeatherForecastProps) {
  return (
    <View style={styles.container}>
      <CustomText
        allowFontScaling={true}
        accessible={true}
        accessibilityLabelText={accessibilityLabelDay}
        text={day}
        size={10}
        color={'#FFFFFF'}
        font={'regular'}
        style={styles.day}
      />
      <CustomText
        allowFontScaling={true}
        accessible={true}
        accessibilityLabelText=" "
        text={temperature}
        size={12}
        color={'#FFFFFF'}
        font={'medium'}
        style={styles.temperature}
      />
      <View
        accessibilityLabel={accessibilityLabelImage}
        style={styles.imageContainer}>
        <Image style={styles.imageStyle} source={icon} />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    marginHorizontal: 5,
  },
  day: {
    marginBottom: 1,
  },
  temperature: {
    marginBottom: 9,
  },
  imageContainer: {
    height: 20,
    justifyContent: 'center',
  },
  imageStyle: {
    height: 20,
    width: 20,
  },
});
