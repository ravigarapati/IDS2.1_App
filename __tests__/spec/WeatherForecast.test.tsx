import 'react-native';
import React from 'react';
import WeatherForecast from '../../src/components/WeatherForecast';
import renderer from 'react-test-renderer';

jest.mock('../../src/components/CustomText', () => 'CustomText');

describe('Weather forecast Component', () => {
  test('Component to be defined', () => {
    const tree = renderer
      .create(
        <WeatherForecast
          day="Today"
          temperature="H:70째/L:60째"
          icon={require('./../../src/assets/images/backgroundsWeather/icons/Cloudy.png')}
        />,
      )
      .toJSON();
    expect(tree).toBeDefined();
  });
  test('Component to be rendered', () => {
    const tree = renderer
      .create(
        <WeatherForecast
          day="Today"
          temperature="H:70째/L:60째"
          icon={require('./../../src/assets/images/backgroundsWeather/icons/Cloudy.png')}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
