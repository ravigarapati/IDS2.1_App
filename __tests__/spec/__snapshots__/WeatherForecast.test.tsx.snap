// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather forecast Component Component to be rendered 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginHorizontal": 5,
    }
  }
>
  <CustomText
    accessibilityLabelText=""
    accessible={true}
    allowFontScaling={true}
    color="#FFFFFF"
    font="regular"
    size={10}
    style={
      Object {
        "marginBottom": 1,
      }
    }
    text="Today"
  />
  <CustomText
    accessibilityLabelText=" "
    accessible={true}
    allowFontScaling={true}
    color="#FFFFFF"
    font="medium"
    size={12}
    style={
      Object {
        "marginBottom": 9,
      }
    }
    text="H:70°/L:60°"
  />
  <View
    accessibilityLabel=""
    style={
      Object {
        "height": 20,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/backgroundsWeather/icons/Cloudy.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
